test/Either/Either.js

```diff
@@ -2,15 +2,124 @@
 
 var throws = require('assert').throws;
 
+var jsc = require('jsverify');
+var R = require('ramda');
+
 var errorEq = require('../utils').errorEq;
 var S = require('../..');
 
 
+//  Identity :: a -> Identity a
+var Identity = function Identity(x) {
+  return {
+    of: Identity,
+    map: function(fn) {
+      return Identity(fn(x));
+    },
+    ap: function(y) {
+      return Identity(x(y));
+    },
+    equals: function(other) {
+      return R.equals(x, other.value);
+    },
+    value: x
+  };
+};
+
+Identity.of = Identity;
+
+//  IdentityArb :: Arbitrary a -> Arbitrary (Identity a)
+var IdentityArb = function(arb) {
+  return arb.smap(Identity, function(i) { return i.value; });
+};
+
+//  MaybeArb :: Arbitrary a -> Arbitrary (Maybe a)
+var MaybeArb = function(arb) {
+  return jsc.oneof(JustArb(arb), jsc.constant(S.Nothing()));
+};
+
+//  JustArb :: Arbitrary a -> Arbitrary (Maybe a)
+var JustArb = function(arb) {
+  return arb.smap(S.Just, function(m) { return m.value; }, R.toString);
+};
+
+//  EitherArb :: Arbitrary a -> Arbitrary b -> Arbitrary (Either a b)
+var EitherArb = function(lArb, rArb) {
+  return jsc.oneof(LeftArb(lArb), RightArb(rArb));
+};
+
+//  LeftArb :: Arbitrary a -> Arbitrary (Either a b)
+var LeftArb = function(arb) {
+  return arb.smap(S.Left, function(e) { return e.value; }, R.toString);
+};
+
+//  RightArb :: Arbitrary a -> Arbitrary (Either b a)
+var RightArb = function(arb) {
+  return arb.smap(S.Right, function(e) { return e.value; }, R.toString);
+};
+
+//  Compose :: Apply f, Apply g
+//          => { of: b -> f b } -> { of: c -> g c }
+//          -> f (g a) -> Compose f g a
+var Compose = function(F, G) {
+  var _Compose = function _Compose(x) {
+    return {
+      constructor: _Compose,
+      map: function(f) {
+        return _Compose(R.map(R.map(f), x));
+      },
+      ap: function(y) {
+        return _Compose(R.ap(R.map(R.ap, x), y.value));
+      },
+      equals: function(other) {
+        return R.equals(x, other.value);
+      },
+      value: x
+    };
+  };
+  _Compose.of = function(x) {
+    return _Compose(F.of(G.of(x)));
+  };
+  return _Compose;
+};
+
```

Comment:
These should be fine to share internally if you'd prefer. In fact, it might be worthwhile investigating reusing the laws defined in the [FantasyLand repo](https://github.com/fantasyland/fantasy-land/blob/master/laws/traversable.js) at some point in the future (though there are some changes I'd want included to allow for passing in arbitrary functions to the laws rather than use `identity` everywhere first)

It might also be a good idea to have the JSVerify arbitrary instances for each type exported somewhere to encourage reuse.